🐛 [WIP] kcp: patch ImageRepository in ClusterConfiguration if it is not set for v1.25

chrischdi opened the pull request:
<!-- please add a icon to the title of this PR (see https://sigs.k8s.io/cluster-api/CONTRIBUTING.md#contributing-a-patch), and delete this line and similar ones -->
<!-- the icon will be either ⚠️ (:warning:, major or breaking changes), ✨ (:sparkles:, feature additions), 🐛 (:bug:, patch and bugfixes), 📖 (:book:, documentation or proposals), or 🌱 (:seedling:, minor or other) -->

**What this PR does / why we need it**:

Patches the ImageRepository variable to `registry.k8s.io` if it was not set explicitly to a value in ClusterConfiguration, when upgrading to v1.25.

For more details and discussion see #6596

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6596

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main

killianmuldoon commented on controlplane/kubeadm/internal/controllers/upgrade.go:
```diff
@@ -74,6 +74,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to update the kubernetes version in the kubeadm config map")
 	}
 
+	// FIXME(chrischdi): use a better semver comparison
```
😆 I think version.ParseMajorMinorPatchTolerant (with or without tolerant) is the best we've got, but there's a few patterns in the codebase.

killianmuldoon replied:
And you can use the version.Compare to get the comparison instead of directly comparing the minor versions

chrischdi replied:
The latter is what the fixme is about :-)

sbueringer replied:
Independent of how we implement it, please ensure that alpha/beta/... versions are also covered :)

chrischdi replied:
yes, will have to because the test will otherwise fail (due to k/k master getting an v1.25 alpha version in the test)

fabriziopandini commented on controlplane/kubeadm/internal/controllers/upgrade.go:
```diff
@@ -74,6 +74,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
+		}
+	}
+
 	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil {
```
Also, wondering if we can combine above change with the current one, so we update the config map only once

fabriziopandini commented on controlplane/kubeadm/internal/controllers/upgrade.go:
```diff
@@ -74,6 +74,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 	}
 
+	// FIXME(chrischdi): use a better semver comparison
+	// FIXME(chrischdi): think about having a better if structure 😅
```
wondering if we can combine above change with the next one, so we update the config map only once 🤔

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main

chrischdi commented:
/test pull-cluster-api-apidiff-main
/test pull-cluster-api-e2e-full-main
/test pull-cluster-api-e2e-informing-ipv6-main
/test pull-cluster-api-e2e-informing-main
/test pull-cluster-api-e2e-workload-upgrade-1-21-1-22-main
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main

chrischdi commented:
Issue seems currently not to be the registry anymore:

KCP got upgraded but first MachineDeployment Machine failed with kube-proxy:
```
Jul 14 15:06:44.894592 k8s-upgrade-and-conformance-zqzlwl-md-0-sfns7-f747cccf5-7q59q kubelet[357]: E0714 15:06:44.894530     357 pod_workers.go:951] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"kube-proxy\" with ImagePullBackOff: \"Back-off pulling image \\\"k8s.gcr.io/kube-proxy:v1.25.0-alpha.2.259_19a22f763770d1\\\"\"" pod="kube-system/kube-proxy-6bs6x" podUID=92247180-cc6d-4702-964c-0b4d1296583f
```

Have to take a look why it still wants to use `k8s.gcr.io` for kube-proxy :-)

chrischdi commented:
> Issue seems currently not to be the registry anymore:
> 
> KCP got upgraded but first MachineDeployment Machine failed with kube-proxy:
> 
> ```
> Jul 14 15:06:44.894592 k8s-upgrade-and-conformance-zqzlwl-md-0-sfns7-f747cccf5-7q59q kubelet[357]: E0714 15:06:44.894530     357 pod_workers.go:951] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"kube-proxy\" with ImagePullBackOff: \"Back-off pulling image \\\"k8s.gcr.io/kube-proxy:v1.25.0-alpha.2.259_19a22f763770d1\\\"\"" pod="kube-system/kube-proxy-6bs6x" podUID=92247180-cc6d-4702-964c-0b4d1296583f
> ```
> 
> Have to take a look why it still wants to use `k8s.gcr.io` for kube-proxy :-)

TLDR: we have to do a similar change at https://github.com/kubernetes-sigs/cluster-api/blob/37a4b2b4c1a377f62c374a1be4a92df90035c329/controlplane/kubeadm/internal/workload_cluster.go#L505

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
`pull-cluster-api-e2e-workload-upgrade-1-24-latest-main` is now green acccording local testing.

/retitle 🐛 kcp: patch ImageRepository in ClusterConfiguration if it is not set for v1.25

chrischdi commented:
/retitle [wip] 🐛 kcp: patch ImageRepository in ClusterConfiguration if it is not set for v1.25

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
/retitle [wip] 🐛 kcp: patch ImageRepository in ClusterConfiguration if it is not set for v1.25

CoreDNS part does not work yet...

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented on controlplane/kubeadm/internal/workload_cluster_coredns.go:
```diff
@@ -195,6 +199,16 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
 	toImageRepository := parsedImage.Repository
 	if clusterConfig.ImageRepository != "" {
 		toImageRepository = strings.TrimSuffix(clusterConfig.ImageRepository, "/")
+	} else if kubernetesVersion.GTE(semver.Version{Major: 1, Minor: 25, Patch: 0, Pre: []semver.PRVersion{{VersionStr: "alpha"}}}) {
```
Note: this would currently cause a rollout even if the image tag does not change for coredns.

Should we prevent updating the repository/registry if the coredns image tag does not change?

fabriziopandini replied:
I'm ok in having a rollout when only the imageRepository changes, this is a valid use case

chrischdi commented on controlplane/kubeadm/internal/workload_cluster_coredns.go:
```diff
@@ -136,9 +138,11 @@ func (w *Workload) UpdateCoreDNS(ctx context.Context, kcp *controlplanev1.Kubead
-	if err := validateCoreDNSImageTag(info.FromImageTag, info.ToImageTag); err != nil {
-		return errors.Wrapf(err, "failed to validate CoreDNS")
+	// Validate the image tag if it was changed.
+	if info.FromImageTag != info.ToImageTag {
```
required due to https://github.com/kubernetes-sigs/cluster-api/pull/6917/files#r922006034 .

Should we prevent updating if the image tag does not change?

chrischdi replied:
The issue was:

if `FromImageTag == ToImageTag`, `validateCoreDNSImageTag` would have returned an error which would result in not updating coredns and failed reconciliation.

chrischdi replied:
I moved the change to the `validateCoreDNSImageTag` func.

If we not allow keeping the image tag the same, we don't allow to change the image repository.

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
/retitle 🐛 kcp: patch ImageRepository in ClusterConfiguration if it is not set for v1.25

cc @sbueringer / @fabriziopandini PTAL :-)

fabriziopandini commented on controlplane/kubeadm/internal/controllers/upgrade.go:
```diff
@@ -74,8 +74,20 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
+	// This ensures that the imageRepository gets patched to registry.k8s.io when upgrading from v1.24 or lower,
+	// but only if there was no imageRespository explicitly set at the KubeadmControlPlanes ClusterConfiguration.
+	// This follows the behavior of `kubeadm upgrade`.
+	if imageRepository == "" && parsedVersion.GTE(semver.Version{Major: 1, Minor: 25, Patch: 0, Pre: []semver.PRVersion{{VersionStr: "alpha"}}}) {
```
Could you kindly introduce a const like e.g. minVerUnversionedKubeletConfig, also, not sure about the Pre part, we don't have it in other cases

fabriziopandini commented on controlplane/kubeadm/internal/workload_cluster.go:
```diff
@@ -500,9 +500,25 @@ func (w *Workload) UpdateKubeProxyImageInfo(ctx context.Context, kcp *controlpla
+	// This ensures that the imageRepositorycontrolplane/kubeadm/internal/workload_cluster_coredns.go gets patched to registry.k8s.io when upgrading from v1.24 or lower,
+	// but only if there was no imageRespository explicitly set at the KubeadmControlPlanes ClusterConfiguration.
+	// This follows the behavior of `kubeadm upgrade`.
+	parsedVersion, err := semver.ParseTolerant(kcp.Spec.Version)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse kubernetes version %q", kcp.Spec.Version)
+	}
```
What about passing version to UpdateKubeProxyImageInfo, so we are consistent with other Update methods like e.g. UpdateCoreDNS etc etc.

fabriziopandini commented on controlplane/kubeadm/internal/workload_cluster.go:
```diff
@@ -500,9 +500,25 @@ func (w *Workload) UpdateKubeProxyImageInfo(ctx context.Context, kcp *controlpla
-	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil &&
-		kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository != "" {
-		newImageName, err = containerutil.ModifyImageRepository(newImageName, kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository)
+	var imageRepository string
```
It seems that the code that computes imageRepository is repeated in two places, what about moving to a func and re-use?

chrischdi replied:
Added a func `ImageRepositoryFromClusterConfig` which gets used in all 3 places.

fabriziopandini commented on controlplane/kubeadm/internal/workload_cluster_coredns.go:
```diff
@@ -195,6 +199,14 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
+		// This ensures that the imageRepository gets patched to registry.k8s.io when upgrading from v1.24 or lower,
+		// but only if there was no imageRespository explicitly set at the KubeadmControlPlanes ClusterConfiguration.
+		// This follows the behavior of `kubeadm upgrade`.
+		if strings.HasPrefix(toImageRepository, oldKubernetesImageRepository) {
+			toImageRepository = KubernetesImageRepository + strings.TrimPrefix(toImageRepository, oldKubernetesImageRepository)
+		}
```
I'm not sure this follow what is in the comment, we are not checking if imageRespository is explicitly set at the KubeadmControlPlanes ClusterConfiguration

chrischdi replied:
Makes now use of `ImageRepositoryFromClusterConfig` which has the wanted behaviour and is way more readable.

chrischdi commented:
Had to fix coredns stuff. Tested locally already, they got to ImagePullBackoff ...

/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
/cherry-pick release-1.2

otherwise there will be no upgrade working for v1.24 -> v1.25 in v1.2

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
rebased 
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

chrischdi commented:
/test pull-cluster-api-e2e-full-main

sbueringer commented on controlplane/kubeadm/internal/workload_cluster.go:
```diff
@@ -82,6 +82,14 @@ var (
 	// NOTE: The following assumes that kubeadm version equals to Kubernetes version.
 	minVerUnversionedKubeletConfig = semver.MustParse("1.24.0")
 
+	// MinKubernetesVersionImageRegistryMigration is first kubernetes version where
+	// images are only available at registry.k8s.io instead of k8s.gcr.io.
+	MinKubernetesVersionImageRegistryMigration = semver.MustParse("1.25.0")
+
+	// NextKubernetesVersionImageRegistryMigration is the next minor version after
+	// migration the image registry to registry.k8s.io.
+	NextKubernetesVersionImageRegistryMigration = semver.MustParse("1.26.0")
+
```
Please unexport the constants.

> images are only available at registry.k8s.io instead of k8s.gcr.io.

That is not correct. Images are still available in both as one is just a redirect to the other. It's more like that's the version where v1.25.0 is the default.

It's just a special case in CAPD tests that we rely on the images being baked in with the correct registry, but I would not mention that here.

chrischdi replied:
:+1: adjusted the comment and also made the vars unexported :-) I did use them used directly in the controller but since adding the `ImageRepositoryFromClusterConfig` func it does not need to get exported anymore 👍

sbueringer commented on controlplane/kubeadm/internal/controllers/upgrade.go:
```diff
@@ -74,8 +75,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
+	// Get the set imageRepository or the correct value if nothing is set and a migration is necessary.
+	imageRepository := internal.ImageRepositoryFromClusterConfig(kcp.Spec.KubeadmConfigSpec.ClusterConfiguration, parsedVersionTolerant)
+
+	if imageRepository != "" {
```
I assume we never have to set the imageRepository to an empty string in the ConfigMap if it was set before?

chrischdi replied:
Hmm good point. We did never do before if `kcp.Spec.KubeadmConfigSpec.ClusterConfiguration` was nil.

So I think we should remove that if, however it seems to cause issues in at least a unit test (nil pointer in https://github.com/kubernetes-sigs/cluster-api/blob/7a533b2f1db794dc8f61ab1021d966051beccf87/controlplane/kubeadm/internal/controllers/scale_test.go#L157).

I would propose restoring the old behaviour by only checking `if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil` here and open an issue for fixing that we don't update the imageRepository if ClusterConfiguration is nil.

WDYT @sbueringer ?

sbueringer commented on controlplane/kubeadm/internal/workload_cluster.go:
```diff
@@ -545,3 +555,28 @@ func yamlToUnstructured(rawYAML []byte) (*unstructured.Unstructured, error) {
+// * clusterConfig.ImageRepository if set.
+// * "registry.k8s.io" if v1.25 <= version < v1.26 to migrate to the new registry
+// * "" otherwise.
+// Beginning with kubernetes v1.25, kubernetes images get published to registry.k8s.io
```
Please update the comment as well. As mentioned above they are available under both

sbueringer commented on controlplane/kubeadm/internal/workload_cluster_coredns.go:
```diff
@@ -47,9 +47,11 @@ const (
-	oldCoreDNSImageName       = "coredns"
-	coreDNSImageName          = "coredns/coredns"
+	// KubernetesImageRepository is the default Kubernetes image repository for build artifacts.
+	KubernetesImageRepository    = "registry.k8s.io"
```
```suggestion
	kubernetesImageRepository    = "registry.k8s.io"
```

Please only create exported constants if they are used by another package. Otherwise we can keep the API surface smaller. (like we did with other constants usually)

chrischdi replied:
same story here: I did use them used directly in the controller but since adding the `ImageRepositoryFromClusterConfig` func it does not need to get exported anymore 👍

sbueringer commented on controlplane/kubeadm/internal/workload_cluster_coredns.go:
```diff
@@ -193,8 +195,17 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
+			if strings.HasPrefix(toImageRepository, oldKubernetesImageRepository) {
+				// Ensure to keep the repository subpaths when patching from oldKubernetesImageRepository to new KubernetesImageRepository.
+				toImageRepository = imageRegistryRepository + strings.TrimPrefix(toImageRepository, oldKubernetesImageRepository)
+			}
```
could it be that we still need the TrimSuffix in this branch?

sbueringer commented on controlplane/kubeadm/internal/workload_cluster_coredns.go:
```diff
@@ -219,7 +230,7 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
 	// Handle the renaming of the upstream image from "k8s.gcr.io/coredns" to "k8s.gcr.io/coredns/coredns"
 	toImageName := parsedImage.Name
-	if toImageRepository == kubernetesImageRepository && toImageName == oldCoreDNSImageName && targetMajorMinorPatch.GTE(semver.MustParse("1.8.0")) {
+	if toImageRepository == oldKubernetesImageRepository && toImageName == oldCoreDNSImageName && targetMajorMinorPatch.GTE(semver.MustParse("1.8.0")) {
```
I wonder if we have to do this for the new registry as well, given that we could have replaced the registry above.

I think the coredns image name migration is still relevant as folks could upgrade to CoreDNS >= v1.8.0 with Kubernetes >= v1.25.0
(most folks probably never set the CoreDNS version and just get the kubeadm default, so they could recognize this relatively late and then upgrade from some very old version to a current one)

chrischdi replied:
I assumed everyone must be on coredns > v1.8.0 due to adding `EndpointSlices` which are only supported by coredns >= v1.8.1.

However for safety we can of course also do this for the new registry :-)

sbueringer replied:
But we only update the ClusterRole to add `EndpointSlices` if the CoreDNS version is actually >= v1.8.1?

I might be missing something but I think the following wouldn't have worked without handling the new registry:
* install cluster with 1.20 (CoreDNS 1.7.0)
* upgrade cluster from 1.20 until 1.25
* set CoreDNS tag to upgrade to CoreDNS >= 1.8.0

chrischdi replied:
Yeah, the other CoreDNS versions do not support EndpointSlices. I agree, which is why I added both versions to the if clause.

sbueringer commented on controlplane/kubeadm/internal/workload_cluster_coredns.go:
```diff
@@ -429,10 +440,9 @@ func validateCoreDNSImageTag(fromTag, toTag string) error {
-	// or if they're the same version, the raw tags should be different (e.g. allow from `v1.17.4-somevendor.0` to `v1.17.4-somevendor.1`).
-	if x := from.Compare(to); x > 0 || (x == 0 && fromTag == toTag) {
-		return fmt.Errorf("toVersion %q must be greater than fromVersion %q", toTag, fromTag)
+	// make sure that the version we're upgrading to is greater or equal to the current one,
```
Why are we changing this? Won't this lead to issues?

chrischdi replied:
See https://github.com/kubernetes-sigs/cluster-api/pull/6917#discussion_r922006323 ,

TLDR: if we don't allow keeping the same tag it would not result in a rollout / changing the image repository for coredns.

chrischdi commented on bootstrap/kubeadm/types/upstreamv1beta1/types.go:
```diff
@@ -107,9 +107,10 @@ type ClusterConfiguration struct {
 
 	// ImageRepository sets the container registry to pull images from.
-	// If empty, `k8s.gcr.io` will be used by default; in case of kubernetes version is a CI build (kubernetes version starts with `ci/` or `ci-cross/`)
+	// If empty, `k8s.gcr.io` or `registry.k8s.io` will be used by default; in case of kubernetes version is a CI build (kubernetes version starts with `ci/` or `ci-cross/`)
```
synced now, but only for this comment. Should we create an issue to do a general sync?

sbueringer replied:
It should be relatively up-to-date. I just did it a few months back when adding some fields.

sbueringer replied:
But you can if you want to. But please add a comment that we intentionally didn't add all fields that exist upstream, as not everything makes sense in a CAPI context (we should probably document that in the copied files)

chrischdi commented:
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main
/test pull-cluster-api-e2e-full-main

fabriziopandini reviewed (commented):
/lgtm
@sbueringer for a last pass

sbueringer commented on controlplane/kubeadm/internal/controllers/upgrade.go:
```diff
@@ -74,8 +75,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to parse kubernetes version %q", kcp.Spec.Version)
+	}
+	// Get the set imageRepository or the correct value if nothing is set and a migration is necessary.
```
```suggestion
	// Get the imageRepository or the correct value if nothing is set and a migration is necessary.
```

sbueringer commented on controlplane/kubeadm/internal/controllers/upgrade.go:
```diff
@@ -74,8 +75,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to update the kubernetes version in the kubeadm config map")
 	}
 
+	// We intentionally only parse major/minor/patch so that the subsequent code
```
Let's move l.78-l.85 into the if

sbueringer commented on controlplane/kubeadm/internal/workload_cluster_coredns.go:
```diff
@@ -136,7 +138,7 @@ func (w *Workload) UpdateCoreDNS(ctx context.Context, kcp *controlplanev1.Kubead
 	}
 
-	// Validate the image tag.
+	// Validate the image tag
```
```suggestion
	// Validate the image tag.
```
nit

sbueringer commented on controlplane/kubeadm/internal/workload_cluster_coredns_test.go:
```diff
@@ -1202,6 +1202,19 @@ func TestGetCoreDNSInfo(t *testing.T) {
 				toImage: "dnsRepo/coredns:1.7.2-foobar.1",
 			},
+			{
+				name: "patches ImageRepository to registry.k8s.io if not on both global and DNS-level set and kubernetesVersion >= v1.25",
```
```suggestion
				name: "patches ImageRepository to registry.k8s.io if it's set on neither global nor DNS-level and kubernetesVersion >= v1.25",
```

chrischdi commented on controlplane/kubeadm/internal/workload_cluster_coredns_test.go:
```diff
@@ -1163,11 +1162,12 @@ func TestGetCoreDNSInfo(t *testing.T) {
+			clusterConfig     *bootstrapv1.ClusterConfiguration
+			kubernetesVersion semver.Version
+			toImage           string
 		}{
```
Had to refactor the tests a bit, but now the PR adds a test cases which test:
* update to v1.25 and v1.6.2 -> v1.8.0: migrate to new registry and at the same time add the `coredns/` prefix
* update to v1.24 and v1.6.2 -> v1.8.0: only add the `coredns/` prefix
* update to v1.26 and v1.6.2 -> v1.8.0: only add the `coredns/` prefix
* update to v1.25 and v1.6.2 -> v1.7.2: only adjust the registry

chrischdi commented:
/test pull-cluster-api-apidiff-main
/test pull-cluster-api-e2e-full-main
/test pull-cluster-api-e2e-informing-ipv6-main
/test pull-cluster-api-e2e-informing-main
/test pull-cluster-api-e2e-workload-upgrade-1-21-1-22-main
/test pull-cluster-api-e2e-workload-upgrade-1-24-latest-main

@sbueringer PTAL 😀

sbueringer commented:
Thank you very much for adding the additional tests.

I think that's really useful as it just gives us a bit more confidence that with all the logic we now have it still behaves as expected.

/lgtm


@fabriziopandini PTAL

sbueringer commented:
@fabriziopandini Cherry-pick into release-1.2? At this point I would assume our goal is to eventually support v1.25 in v1.2.x if possible. This would mean we have to cherry-pick all relevant v1.25 PRs for the time being.

fabriziopandini commented:
@chrischdi release 1-1 is technically out of support and thus we should encourage users to move to 1.2, so I'm for removing the test.

Also, side question, as far as you remember, have we updated the support matrix pages with 1.25 being supported starting from the next patch release?

chrischdi commented:
> @chrischdi release 1-1 is technically out of support and thus we should encourage users to move to 1.2, so I'm for removing the test.
> 
> Also, side question, as far as you remember, have we updated the support matrix pages with 1.25 being supported starting from the next patch release?

We have not added 1.25 to the matrix page yet.
I think we have to wait until `v1.25` is released and we added `capi-e2e-release-1-2-1-24-1-25` for v1.2.

Both (version matrix and adding the periodics) is already tracked in:

- #6661

chrischdi commented:
I will file a PR then to remove https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-1.1#capi-e2e-release-1-1-1-24-latest

sbueringer commented:
+1 We should only claim v1.25 support once we have green tests for v1.25.0
